test/loading.jl

```diff
@@ -765,8 +765,25 @@ end
 @testset "`Base.project_names` and friends" begin
     # Some functions in Pkg assumes that these tuples have the same length
     n = length(Base.project_names)
-    @test length(Base.manifest_names) == n
+    @test length(Base._manifest_names) == n
+    @test !haskey(ENV, "JULIA_MANIFEST_NAME")
```

Comment:
Instead of hoping that the `JULIA_MANIFEST_NAME` environment variable was not set in the user's environment when they ran the tests, I would instead do:

```julia
withenv("JULIA_MANIFEST_NAME" => nothing) do
    @test length(Base.manifest_names()) == n
    ...
    ...
end
```